vault/mount.go

```diff
@@ -323,6 +328,7 @@ type MountEntry struct {
 	Tainted               bool              `json:"tainted,omitempty"`                 // Set as a Write-Ahead flag for unmount/remount
 	MountState            string            `json:"mount_state,omitempty"`             // The current mount state.  The only non-empty mount state right now is "unmounting"
 	NamespaceID           string            `json:"namespace_id"`
+	LastMounted           string            `json:"last_mounted,omitempty"` // The last version of Vault with which this entry was mounted.
```

Comment:
Rather than maintaining this in every mount entry in storage, could we simplify by calculating `isNonPatchUpdate` inside `handleVersionTimeStamps`, and then passing that as a boolean parameter to `setupMounts` and `setupCredentials`?

Reply:
My initial approach was actually similar (except we need to persist the "mounted" status).

We need to record and persist `LastMounted` so that we can stop Vault and tell whether it was successfully mounted in the past. To achieve this, I added a storage entry in the version store, but that led to all sorts of test failures...

I think I prefer the current approach, since it provides some useful information about each MountEntry and seems like a more appropriate place to store this information.

Do you see any potential problems with this approach, that I could be missing?

Reply:
I suppose I could still use that approach and add a new storage entry (outside of the version store), but this feels like potentially useful information to store in each mount entry...

Reply:
@tomhjp what are your thoughts? Is this a showstopper, or are you alright with moving forward with this approach?

Reply:
That makes sense re: the possibility of a failed unseal post-upgrade, so the natural next place I'd go is what you started with of adding a single additional storage entry of some form to mark a completed upgrade.

My concern with adding it to the mount entry storage is that it's a much trickier proposition because of the many places we upgrade/update mount storage, and I personally don't see compelling immediate use-cases for breaking that data down per-mount. However, my concerns aren't big enough to be a blocker as long as there's good test coverage.